lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -413,6 +413,10 @@ def update_owner_and_excludes
 
   # @!visibility private
   def git_with_identity(*args)
+    if @resource.value(:http_proxy)
+      ENV['http_proxy'] = @resource.value(:http_proxy)
+      ENV['https_proxy'] = @resource.value(:http_proxy)
+    end
```

Comment:
You must restore these back to their previous values after calling git / git-helper below, otherwise they will affect other components in the puppet run.